### PR TITLE
Add Azure Blob storage backend

### PR DESCRIPTION
**Pull request by philliphoff**
Adds a backend implementation based on Azure Blob Storage. Its use and implementation is based on that for Amazon S3; I'm not a Go developer so I've tried to match it as closely as possible. By all means provide feedback on syntax, style, error handling, etc.

I wasn't able to get the acceptance tests running; any pointers related to their proper setup would be helpful.

Associated with #24.

### CODE REVIEW

**Comment by jdolitsky** [comment]
@philliphoff This is awesome! Thank you, thank you, thank you!

Need to sign up for Azure and try this out. Will provide feedback asap.

File: cmd/chartmuseum/main.go [hunk 1]
```diff
@@ -231,6 +242,21 @@ var cliFlags = []cli.Flag{
...
+	},
+	cli.StringFlag{
+		Name:   "storage-azure-key",
+		Usage:  "key to store charts for --storage-azure-key",
+		EnvVar: "STORAGE_AZURE_KEY",
+	},
+	cli.StringFlag{
+		Name:   "storage-azure-container",
+		Usage:  "container to store charts for --storage-azure-container",
+		EnvVar: "STORAGE_AZURE_KEY",
```
Comment (davidovich) [inline cmd/chartmuseum/main.go]: I believe this should be STORAGE_AZURE_CONTAINER

Comment (philliphoff) [reply cmd/chartmuseum/main.go]: Correct, thanks!

**Comment by philliphoff** [comment]
Apologies, but there's some red tape on my side that I need to wade through before I can allow this PR to be committed, so closing this until that's done (but keeping my fork/branch).

**Comment by jdolitsky** [comment]
@philliphoff sounds good, sorry it’s taken so long on our end. Will use your branch to continue testing

**Pull request closed without merging**